src/core/weights/nodeWeights/nodeWeights.js

```diff
@@ -0,0 +1,53 @@
+// @flow
+
+import {type NodeAddressT} from "../../graph";
+import {nodeWeightEvaluator} from "../../algorithm/weightEvaluator";
+import {orElse as either} from "../../../util/null";
+import {
+  type NodeWeightsT,
+  type NodeOperator,
+  type NodeWeight,
+  empty,
+  copy,
+  merge,
+} from "./nodeWeightsT";
+
+export type NodeWeightsI = {|
+  fold: () => NodeWeightsT,
+  getNodeWeight: (NodeAddressT) => NodeWeight,
+  setNodeWeight: (NodeAddressT, NodeWeight) => NodeWeightsI,
+  merge: (
+    nws: $ReadOnlyArray<NodeWeightsI>,
+    nodeResolver?: NodeOperator
+  ) => NodeWeightsI,
+  copy: () => NodeWeightsI,
+|};
+
+/**
+ * NodeWeights lifts NodeWeightsT into a context that
+ * lazily handles the creation of a NodeWeightEvaluator,
+ * which must be constructed to retrieve a node's weight.
+ *
+ * In addition, this offers a simpler interface to working
+ * with weights, as NodeWeightsT transformations are abstracted
+ * into functions on this object.
+ */
+export const NodeWeights = (
+  nodeWeightsT: NodeWeightsT = empty()
+): NodeWeightsI => {
+  let nwe;
+  return {
+    fold: () => nodeWeightsT,
+    getNodeWeight: (address: NodeAddressT): NodeWeight => {
+      nwe = either(nwe, nodeWeightEvaluator(nodeWeightsT));
+      return nwe(address);
```

Comment:
Why are we lazily defining the node and edge weight evaluators on the fly? If this is provided by the algorithm module and only used by the algorithm module why not just definitively assign it?

Reply:
It's a sort of a gorilla/banana dynamic here.  If we're just using Weights to modify data or merge or copy Weights data, we don't need to carry along this trie with us.

Reply:
I would just load it if the lazily creation was complicated, but I think the lazy version is simple enough.